File: App/Model/Voucher.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App.Model
{
    public class Voucher : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler? PropertyChanged;

        public string Code { get; set; } // Mã giảm giá
        public int Quantity { get; set; } // Số lượng voucher
        public string Note { get; set; } // Ghi chú (nếu có)
        public decimal MinOrder { get; set; } // Giá trị đơn hàng tối thiểu
        public decimal DiscountValue { get; set; } // Giá trị giảm giá
        public DateTimeOffset? StartDate { get; set; } // Dùng nullable để tránh lỗi
        public DateTimeOffset? EndDate { get; set; } // Dùng nullable để tránh lỗi

        public Voucher() { }

        public Voucher(string code, DateTimeOffset? startDate, DateTimeOffset? endDate, int quantity, decimal minOrder, decimal discountValue, string note)
        {
            Code = code;
            StartDate = startDate;
            EndDate = endDate;
            Quantity = quantity;
            MinOrder = minOrder;
            DiscountValue = discountValue;
            Note = note;
        }
    }
}

File: BE/src/models/Voucher.js
const mongoose = require('mongoose');

const VoucherSchema = new mongoose.Schema({
    Code: { type: String, required: true, unique: true }, // Mã giảm giá
    Quantity: { type: Number, required: true }, // Số lượng voucher
    Note: { type: String, required: false }, // Ghi chú (nếu có)
    MinOrder: { type: Number, required: true }, // Giá trị đơn hàng tối thiểu
    DiscountValue: { type: Number, required: true }, // Giá trị giảm giá
    StartDate: { type: Date, required: false }, // Ngày bắt đầu (nullable)
    EndDate: { type: Date, required: false } // Ngày kết thúc (nullable)
});

const Voucher = mongoose.model('voucher', VoucherSchema, 'voucher');
module.exports = Voucher;
